src/modules/qt/filter_qresize.cpp

```diff
@@ -0,0 +1,344 @@
+/*
+ * filter_resize.c -- resizing filter
+ * Copyright (C) 2003-2014 Meltytech, LLC
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+#include "common.h"
+#include <framework/mlt.h>
+#include <framework/mlt_log.h>
+#include <QPainter>
+#include <QPainterPath>
+#include <QString>
+
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <math.h>
+
+static uint8_t *resize_alpha( uint8_t *input, int owidth, int oheight, int iwidth, int iheight, uint8_t alpha_value )
+{
+	uint8_t *output = NULL;
+
+	if ( input != NULL && ( iwidth != owidth || iheight != oheight ) && ( owidth > 6 && oheight > 6 ) )
+	{
+		uint8_t *out_line;
+		int offset_x = ( owidth - iwidth ) / 2;
+		int offset_y = ( oheight - iheight ) / 2;
+		int iused = iwidth;
+
+		output = (uint8_t*)mlt_pool_alloc( owidth * oheight );
+		memset( output, alpha_value, owidth * oheight );
+		offset_x -= offset_x % 2;
+
+		out_line = output + offset_y * owidth;
+		out_line += offset_x;
+
+		// Loop for the entirety of our output height.
+		while ( iheight -- )
+		{
+			// We're in the input range for this row.
+			memcpy( out_line, input, iused );
+
+			// Move to next input line
+			input += iwidth;
+
+			// Move to next output line
+			out_line += owidth;
+		}
+	}
+
+	return output;
+}
+
+static void resize_image( uint8_t *output, int owidth, int oheight, uint8_t *input, int iwidth, int iheight, int bpp, mlt_image_format format, uint8_t alpha_value )
+{
+	// Calculate strides
+	int istride = iwidth * bpp;
+	int ostride = owidth * bpp;
+	int offset_x = ( owidth - iwidth ) / 2 * bpp;
+	int offset_y = ( oheight - iheight ) / 2;
+	uint8_t *in_line = input;
+	uint8_t *out_line;
+	int size = owidth * oheight;
+	uint8_t *p = output;
+
+	// Optimisation point
+	if ( output == NULL || input == NULL || ( owidth <= 6 || oheight <= 6 || iwidth <= 6 || iheight <= 6 ) )
+	{
+		return;
+	}
+	else if ( iwidth == owidth && iheight == oheight )
+	{
+		memcpy( output, input, iheight * istride );
+		return;
+	}
+
+	if ( format == mlt_image_rgb24a )
+	{
+#if QT_VERSION >= 0x050200
+		QImage src(iwidth, owidth, QImage::Format_RGBA8888);
```

Comment:
There is no need to use the the convert_* functions from common.cpp since you have already guarded with with a QT version check that removes the need for backwards compatibility. The way this is written, src will be initialized twice. Once on line 93, and then again in convert_mlt_to_qimage().

I would suggest something like:

```
QImage src(input, iwidth, iheight, QImage::Format_RGBA8888);
QImage dest(output, owidth, oheight, QImage::Format_RGBA8888 );
dest.fill(Qt::transparent);
QPainter painter(&dest);
painter.drawImage(QPoint((owidth - iwidth) / 2, (oheight - iheight) / 2), src);
painter.end();
```

Reply:
Please test this code (whichever approach) with a variety of input and output sizes to avoid a regression, which I just found like this https://github.com/mltframework/mlt/commit/27361ebadedaa299a97b1a638fd7bc5675503a6c?diff=unified#r39869941